### PR TITLE
Fix label molecule

### LINKED ISSUE #222: ForceField.create_openmm_system(topology) should read the box vectors from topology
Currently, they are ignored so if `create_openmm_system(..., default_box_vectors=None)` the resulting `System` object will have arbitrary box vectors of 2nm per side.

Relatedly, now that the OFF `Topology` comes with box vectors, do we still want to keep the `default_box_vectors` argument to overwrite them? Or do we want the user to set `topology.box_vectors` before passing it to `create_openmm_system()`?

### PR DESCRIPTION
**Pull request by j-wags**
* Rewrites `label_molecule` example to work using ipython
* Fixes #222

### CODE REVIEW

**Comment by j-wags** [comment]
This pull request **fixes 1 alert** when merging a0e3bc4a855a64cd50bc97d9f074b7c9d1b666f5 into 2e86a5c9a37a0334137754911b21b6864ec97152 - [view on LGTM.com](https://lgtm.com/projects/g/openforcefield/openforcefield/rev/pr-9b1bee7a0fac8bbe987a754b0f57dc08471ebf46)

**fixed alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by jchodera** [comment]
I think we want to stress that this API isn't final, and that we will likely come up with a better object model representation for visualizing assigned parameters in the near future.

**Comment by j-wags** [comment]
This pull request **fixes 1 alert** when merging 787209f155be0c562292972e74ee47a33f2ff7b2 into 2e86a5c9a37a0334137754911b21b6864ec97152 - [view on LGTM.com](https://lgtm.com/projects/g/openforcefield/openforcefield/rev/pr-43e311af6d51193e7b3fed1750e11b49d3faafaa)

**fixed alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: openforcefield/typing/engines/smirnoff/forcefield.py [hunk 1]
```diff
@@ -828,9 +821,8 @@ def create_openmm_system(self,
         # Create an empty OpenMM System
         system = openmm.System()
 
-        # Set periodic boundary conditions if specified
-        if default_box_vectors is not None:
-            system.setDefaultPeriodicBoxVectors(default_box_vectors)
+        # Set periodic boundary conditions from topology
+        system.setDefaultPeriodicBoxVectors(topology.box_vectors)
```
Comment (andrrizzi) [inline openforcefield/typing/engines/smirnoff/forcefield.py]: I'm not sure this works when `topology.box_vectors` is `None`.

File: openforcefield/typing/engines/smirnoff/forcefield.py [hunk 2]
```diff
@@ -930,12 +915,15 @@ def label_molecules(self, topology, verbose=False):
...
         # Loop over molecules and label
         molecule_labels = list()
         for molecule_idx, molecule in enumerate(topology.reference_molecules):
+            top_mol = Topology.from_molecules([molecule])
             current_molecule_labels = dict()
-            for parameter_handler in self._parameter_handlers.values():
-                matches = parameter_handler.find_matches(molecule)
-                molecule_labels[molecule_idx][parameter_handler.name] = matches
+            for tag, parameter_handler in self._parameter_handlers.items():
+                matches = parameter_handler.get_matches(top_mol)
```
Comment (andrrizzi) [inline openforcefield/typing/engines/smirnoff/forcefield.py]: I think I renamed `get_matches` to `find_matches` in the previous PR.

File: openforcefield/typing/engines/smirnoff/forcefield.py (hunk 1 above)
Comment (j-wags) [reply openforcefield/typing/engines/smirnoff/forcefield.py]: Good catch. It looks like we need to ensure that our setting of nonbonded methods takes into account that we don't have box vectors, and OpenMM will ignore them after that. Put the "if" statement back in 4f8d950

**Comment by j-wags** [comment]
This pull request **fixes 1 alert** when merging 4f8d9500300bc08c720e7fd681550e8724bdb9db into 2e86a5c9a37a0334137754911b21b6864ec97152 - [view on LGTM.com](https://lgtm.com/projects/g/openforcefield/openforcefield/rev/pr-0937b952c2900663ee8b18c9d83f747b3f412682)

**fixed alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: openforcefield/typing/engines/smirnoff/forcefield.py (hunk 2 above)
Comment (j-wags) [reply openforcefield/typing/engines/smirnoff/forcefield.py]: Thanks. Fixed in be96018

**Comment by j-wags** [comment]
This pull request **fixes 1 alert** when merging be96018decf840f2fdf0ce82c0cfeb2030d1d0d6 into e1d71c44c8aa08fb55e78ad5cf6a986891f25ee4 - [view on LGTM.com](https://lgtm.com/projects/g/openforcefield/openforcefield/rev/pr-3887d22a96ec8d52969dff65d6478c9534f49176)

**fixed alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by j-wags** [comment]
This pull request **fixes 1 alert** when merging 7b985ffa719ec06279c93b14aeb2467eefca21a6 into e1d71c44c8aa08fb55e78ad5cf6a986891f25ee4 - [view on LGTM.com](https://lgtm.com/projects/g/openforcefield/openforcefield/rev/pr-7a320ad3b5e682a3aec9bbecf509d8b154cf64f6)

**fixed alerts:**

* 1 for &#39;import \*&#39; may pollute namespace

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**